sim/core/bulksim.go

```diff
@@ -184,7 +186,29 @@ func (b *bulkSimRunner) Run(pctx context.Context, progress chan *proto.ProgressM
 		}
 		substitutedRequest, changeLog := createNewRequestWithSubstitution(b.Request.BaseSettings, sub, b.Request.BulkSettings.AutoEnchant)
 		if isValidEquipment(substitutedRequest.Raid.Parties[0].Players[0].Equipment) {
+			// Need to sim base dps of gear loudout
 			validCombos = append(validCombos, singleBulkSim{req: substitutedRequest, cl: changeLog, eq: sub})
+			// Todo(Netzone-GehennasEU): Make this its own step?
+			if !b.Request.BulkSettings.SimTalents {
+
+			} else {
+				var talentsToSim = b.Request.BulkSettings.GetTalentsToSim()
+
+				if len(talentsToSim) > 0 {
+					for _, talent := range talentsToSim {
+						sr := goproto.Clone(substitutedRequest).(*proto.RaidSimRequest)
+						cl := *changeLog
+						if sr.Raid.Parties[0].Players[0].TalentsString == talent.TalentsString && reflect.DeepEqual(talent.Glyphs, sr.Raid.Parties[0].Players[0].Glyphs) {
```

Comment:
It's been quite a while since I have written / worked on the sim, so please take the comment(s) with a grain of salt. Does this first comparison depend on the order of the specified talent loadouts, or does this always remove duplicates? I.e. if the standard settings use loadout B and the bulk settings specify loadouts [A, B] (in that order), does this comparison actually prevent the duplicate sim?

For the second comparison, can you use proto.Equal? AFAIK reflect.DeepEqual depends on implementation details on protoc for Go, and the generated structs hold things like mutexes and other fields that you probably don't want to compare. https://pkg.go.dev/google.golang.org/protobuf/proto#Equal

Reply:
Your first part I'm unsure of, I normally write .NET to be honest, but it does prevent the duplicate sim in the GUI condition. When substitutedRequest is created it uses the base request talents, so whatever is your base simRequest if I understand it correctly.

I changed the DeepEqual to proto.Equal, will verify that it works.

Reply:
`proto.Equal` does work yep, you're right that it seems like a better comparison.